### PR TITLE
feat: allow any type of processor classes

### LINKED ISSUE #44: Ease adding of new processors
The existing processors are hardcoded at various places like https://github.com/pimcore/web-to-print-bundle/blob/4f46683fff81c6b88d537251a47d6da107a98720/src/Processor.php#L39
https://github.com/pimcore/web-to-print-bundle/blob/4f46683fff81c6b88d537251a47d6da107a98720/src/Processor.php#L47

Otherwise we could easily use a new Processor Class (using dompdf or wkhtmltopdf) with this bundle by simply extending Pimcore\Bundle\WebToPrintBundle\Processor and implementing the functions.

Currently we'd need to fork the complete repository as far as I can see. What do you think?

I'd be happy to provide a PR (need to touch all processor definitions code anyway to add a new one).

### PR DESCRIPTION
**Pull request by Cruiser13**
Allows any processor classes as discussed in #44

Tagged services and a service locator would be the better way but overpowered for this use case I think. Let me know your thoughts.

Thank you for considering the PR.

### CODE REVIEW

**Comment by Cruiser13** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Cruiser13** [comment]
@kingjia90 I see that a new release has been created without this PR. Are there any changes to be made?

**Comment by kingjia90** [comment]
@Cruiser13 Unfortunately this is PR was not being reviewed yet and not in the main pipeline.

The changes looks fine so far, at least as quick fix, but maybe lacks a way to provide the settings by GUI eg.
https://github.com/pimcore/web-to-print-bundle/blob/70b7474d17b7b792cd6f52e45100aab58b5a5f7f/public/js/settings.js#L342-L353

File: src/Processor.php [hunk 1]
```diff
@@ -40,12 +40,25 @@ public static function getInstance(): PdfReactor|Gotenberg|Chromium|Processor
...
+        if($config['generalTool'] == 'pdfreactor'){
+            return new PdfReactor();
+        }
+        elseif($config['generalTool'] == 'chromium'){
+            return new Chromium();
+        }
+        elseif($config['generalTool'] == 'gotenberg'){
+            return new Gotenberg();
+        }
+        else{
```
Comment (kingjia90) [inline src/Processor.php:52]:
```suggestion
```

if we change https://github.com/pimcore/web-to-print-bundle/blob/70b7474d17b7b792cd6f52e45100aab58b5a5f7f/public/js/settings.js#L335-L337 to 
```
["PdfReactor", "PDFreactor"], 
["Chromium", "Chromium"], 
["Gotenberg", "Gotenberg Chromium"], 
```
we could rely on the `class_exists`, but overall `class_exits` shoulld be avoided if possible

Comment (Cruiser13) [reply src/Processor.php:52]: @kingjia90 I thought we might have class_exists as fallback only so I left the other code untouched here. I'm also not sure if we'd need to have a migration migrating the configs from gotenberg to Gotenberg in your case.

**Comment by Cruiser13** [comment]
> @Cruiser13 Unfortunately this is PR was not being reviewed yet and not in the main pipeline.
> 
> The changes looks fine so far, at least as quick fix, but maybe lacks a way to provide the settings by GUI eg.
> 
> https://github.com/pimcore/web-to-print-bundle/blob/70b7474d17b7b792cd6f52e45100aab58b5a5f7f/public/js/settings.js#L342-L353

Thanks for letting me know. Since we can not know which settings the custom processor has I don't see a good way to add them here. And you can override the JS in your custom processor anyway. We re-implemented the wkhtmltopdf processor with these changes and simply supplied the web2print options in the class. Works fine.

**Comment by Cruiser13** [comment]
@kingjia90 what do you think?

**Pull request merged**